returnn/frontend/_backend.py

```diff
@@ -1293,6 +1293,22 @@ def tensor_array_get_item(cls, tensor_array: TensorArrayType, index: Union[int,
             return tensor_array[index]
         raise NotImplementedError
 
+    @classmethod
+    def gradient_checkpoint(cls, fn: Callable[[Tensor], Tensor], *args: List[Tensor]) -> Tensor:
```

Comment:
I think the typing is wrong in multiple ways. If you accept multiple args for `fn`, shouldn't it be `Callable[[Tensor, ...], Tensor]` or so? And also, I think the type for `args` is just `Tensor` (I think you specify the type of individual elements).